tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,497 @@
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Payload Type</td>
+    <td>uint32</td>
+    <td>
+      Set to <strong>value 6</strong> to denote a <i>TIP-20 Transaction Payload</i>.
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Essence <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Transaction Essence</summary>
+        <blockquote>
+          Describes the essence data making up a transaction by defining its inputs, outputs and an optional payload.
+        </blockquote>
+        <table>
+          <tr>
+            <td><b>Name</b></td>
+            <td><b>Type</b></td>
+            <td><b>Description</b></td>
+          </tr>
+          <tr>
+            <td>Transaction Type</td>
+            <td>uint8</td>
+            <td>
+              Set to <strong>value 1</strong> to denote a <i>TIP-20 Transaction Essence</i>.
+            </td>
+          </tr>
+          <tr>
+            <td>Network ID</td>
+            <td>uint64</td>
+            <td>
+              The unique value denoting whether the message was meant for mainnet, testnet, or a private networks.
+            </td>
+          </tr>
+          <tr>
+            <td>Inputs Count</td>
+            <td>uint16</td>
+            <td>The number of input entries.</td>
+          </tr>
+          <tr>
+            <td valign="top">Inputs <code>anyOf</code></td>
+            <td colspan="2">
+              <details>
+                <summary>UTXO Input</summary>
+                <blockquote>
+                  Describes an input which references an unspent transaction output to consume.
+                </blockquote>
+                <table>
+                  <tr>
+                    <td><b>Name</b></td>
+                    <td><b>Type</b></td>
+                    <td><b>Description</b></td>
+                  </tr>
+                  <tr>
+                    <td>Input Type</td>
+                    <td>uint8</td>
+                    <td>
+                      Set to <strong>value 0</strong> to denote an <i>UTXO Input</i>.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td>Transaction ID</td>
+                    <td>ByteArray[32]</td>
+                    <td>The BLAKE2b-256 hash of the transaction payload containing the referenced output.</td>
+                  </tr>
+                  <tr>
+                    <td>Transaction Output Index</td>
+                    <td>uint16</td>
+                    <td>The output index of the referenced output.</td>
+                  </tr>
+                </table>
+              </details>
+            </td>
+          </tr>
+          <tr>
+            <td>Inputs Commitment</td>
+            <td>ByteArray[32]</td>
+            <td>
+              BLAKE2b-256 hash of the serialized outputs referenced in <i>Inputs</i> by their <i>Output IDs</i> (<i>Transaction ID</i> || <i>Transaction Output Index</i>).
```

Comment:
```suggestion
              BLAKE2b-256 hash of the BLAKE2b-256 hashes of the serialized outputs referenced in <i>Inputs</i> by their <i>Output IDs</i> (<i>Transaction ID</i> || <i>Transaction Output Index</i>).
```

Reply:
I would like to merklelize the input commitment, so we would have the chance to compress data in the ledger on nodes.

With this change a node owner could decide to remove the content of the metadata fields of the output in his own node database. If the node hashes the output before removing the data, and stores the hash, the check for correct input commitment would still pass.